pkg/scheduler/periodic.go

```diff
@@ -57,3 +57,26 @@ func (s *Periodic) ScheduleEvery(interval time.Duration, name string, fn func(ct
 		}
 	}()
 }
+
+type Runner interface {
+	Run(ctx context.Context) error
+	Name() string
+}
+
+func RunForever(ctx context.Context, interval time.Duration, runners ...Runner) {
```

Comment:
Can you add docs for how this is different than the existing `ScheduleEvery` function?  Looks like this version will run on every node vs `ScheduleEvery` will only run on the leader.  Both need docs actually.